### DISCUSSION TITLE
Is support for authenticated requests to be added anytime soon?

### DISCUSSION THREAD

**Discussion by sajalverma17** [Rooyca/obsidian-api-request#32] [General]
I see that this is in the TODO of the Readme, just wanted to gauge the interest in and plans for developing and releasing this feature. 

An idea would be to allow users to provide `client_id` and `client_secret` in plugin settings page. 

My usecase is I wanted my notes to pull data from Fitbit. I haven't developed an Obsidian plugin before, but considered developing a new plugin myself for this 🤓

Fortunately found this plugin which nearly does what I want.

**Comment by Rooyca** [comment]
This should be already implemented. You can check [the documentation](https://rooyca.github.io/obsidian-api-request/en/).

**Comment by sajalverma17** [comment]
I can see that I can pass bearer token in the header, but those are usually short-lived. Is there plans to add support for token fetching/refreshing implicitly?

**Reply by Rooyca** [reply]
No, at least not at the moment. Although PR's are always welcome.
